Model/Adapters/ArrayAdapter.php

```diff
@@ -3,106 +3,87 @@
  * Copyright © 2016 FireGento e.V. - All rights reserved.
  * See LICENSE.md bundled with this module for license details.
  */
+
 namespace FireGento\FastSimpleImport\Model\Adapters;
-class ArrayAdapter extends \Magento\ImportExport\Model\Import\AbstractSource
+
+use Magento\ImportExport\Model\Import\AbstractSource;
+
+class ArrayAdapter extends AbstractSource
 {
-    /**
-     * @var int
-     */
-    protected $_position = 0;
+    private int     $position = 0;
+    protected array $array    = [];
 
-    /**
-     * @var array The Data; Array of Array
-     */
-    protected $_array = array();
+    public function __construct(
+        array $data
+    ) {
+        $this->array = $data;
+        $colnames = array_keys($this->current());
+        parent::__construct($colnames);
+    }
 
     /**
      * Go to given position and check if it is valid
      *
-     * @throws \OutOfBoundsException
      * @param int $position
      * @return void
+     * @throws \OutOfBoundsException
      */
     public function seek($position)
     {
-        $this->_position = $position;
+        $this->position = $position;
 
         if (!$this->valid()) {
-            throw new \OutOfBoundsException("invalid seek position ($position)");
+            throw new \OutOfBoundsException("Invalid seek position ($position)");
         }
     }
 
-    /**
-     * ArrayAdapter constructor.
-     * @param array $data
-     */
-    public function __construct($data)
-    {
-        $this->_array = $data;
-        $this->_position = 0;
-        $colnames = array_keys($this->current() );
-        parent::__construct($colnames);
-    }
-
     /**
      * Rewind to starting position
-     *
-     * @return void
      */
-    public function rewind()
+    public function rewind(): void
     {
-        $this->_position = 0;
+        $this->position = 0;
     }
 
     /**
      * Get data at current position
-     *
-     * @return mixed
      */
-    public function current()
+    public function current(): array
     {
-        return $this->_array[$this->_position];
+        return $this->array[$this->position];
     }
 
     /**
      * Get current position
-     *
-     * @return int
      */
-    public function key()
+    public function key(): int
     {
-        return $this->_position;
+        return $this->position;
     }
 
     /**
      * Set pointer to next position
-     *
-     * @return void
      */
-    public function next()
+    public function next(): void
     {
-        ++$this->_position;
+        ++$this->position;
     }
 
     /**
      * Is current position valid?
-     *
-     * @return bool
      */
-    public function valid()
+    public function valid(): bool
     {
-        return isset($this->_array[$this->_position]);
+        return isset($this->array[$this->position]);
     }
 
     /**
-     * Column names getter.
-     *
-     * @return array
+     * @return array|string[]
```

Comment:
For consistency you could also have removed the return type annotation after adding the type hint like you did for the other methods before this.
Also removing the description and replace it with the return type seems a little bit odd, compared to the other DocBlocks.

Reply:
I did it this way because:

- The return type annotation `string[]` is more detailed than what I can set as a hard return type. As a developer, I like the additional information that it is an array **of strings**.
- The description "Column names getter" doesn't give more information than the function name. 

As a rule of thumb, I only add information to the doc block which is not clear from the method declaration.